### DISCUSSION TITLE
Can Authentik Create Users in an External LDAP Source?.

### DISCUSSION THREAD

**Discussion by SirWobbyTheFirst** [goauthentik/authentik#11130] [Q&A]
I have been experimenting with Authentik for my Home Lab and have successfully setup replication with Active Directory and can update passwords, however I was wondering if it is possible to create users in an LDAP source using Authentik? I can see that this appears to be something Keycloak can do, it can use LDAP to create a user in an LDAP source. I understand the account used for Authentik to authenticate against AD will need additional permissions but I want to know if it is even possible in the first place.

If it is possible, how would you configure that? I have setup the user creation flows in Authentik but when it created the user, it just created it in Authentik's own users list. I understand there are security implications for this but it is a test environment and I just want to see if it is possible.

Failing that, could Authentik act as a central LDAP proxy of sorts, in that applications connect to Authentik's LDAP outpost as an LDAP server and the users returned by Authentik would also include those returned by the downstream LDAP source from AD? So for example, I have Joe Bloggs created in Authentik and Sue Bloggs created in Active Directory, I could put in Joe Bloggs as the username and Authentik would validate that user's password but then if I put in Sue Bloggs as the username, Authentik sends that further down the line to AD as normal?

**Comment by joachimtingvold** [comment]
> So for example, I have Joe Bloggs created in Authentik and Sue Bloggs created in Active Directory, I could put in Joe Bloggs as the username and Authentik would validate that user's password but then if I put in Sue Bloggs as the username, Authentik sends that further down the line to AD as normal?

Does this not already work like this? I thought all user info (including passwords) were synced to Authentik? I have never tested this in detail, but I would expect synced users to work even if you shut down the external LDAP-servers you're syncing with? But maybe I have wrongly assumed this?
